backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/projection/impl/ElasticsearchProjectionTypeKeys.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.elasticsearch.search.projection.impl;
+
+import static org.hibernate.search.engine.search.projection.spi.ProjectionTypeKeys.key;
+
+import org.hibernate.search.engine.search.common.spi.SearchQueryElementTypeKey;
+import org.hibernate.search.engine.search.projection.spi.CompositeProjectionBuilder;
+
+public final class ElasticsearchProjectionTypeKeys {
+
+	private ElasticsearchProjectionTypeKeys() {
+	}
+
+	public static final SearchQueryElementTypeKey<CompositeProjectionBuilder> JSON_HIT = key( "JSON hit" );
+	public static final SearchQueryElementTypeKey<CompositeProjectionBuilder> SOURCE = key( "source" );
```

Comment:
Since it's a key, it'd stick to a simpler character set, say lowercase alphanumerics + hyphens like we do elsewhere?

Also since you don't use the builder, you might as well hide its type (`CompositeProjectionBuilder` is incorrect anyway):

```suggestion
	public static final SearchQueryElementTypeKey<?> JSON_HIT = key( "json-hit" );
	public static final SearchQueryElementTypeKey<?> SOURCE = key( "source" );
```